cpp/src/parquet/encoding_benchmark.cc

```diff
@@ -199,6 +199,107 @@ static void BM_PlainDecodingFloat(benchmark::State& state) {
 
 BENCHMARK(BM_PlainDecodingFloat)->Range(MIN_RANGE, MAX_RANGE);
 
+static void BM_PlainEncodingSpacedBoolean(benchmark::State& state) {
+  const int num_values = state.range(0);
+  bool* values = new bool[num_values];
+  // Fixed half spaced pattern
+  std::vector<uint8_t> valid_bits(arrow::BitUtil::BytesForBits(num_values), 0b10101010);
```

Comment:
Should use `std::vector` instead.

Reply:
Seems can't, the PutSpaced only accept T* input, but Put has std::vector<T> input.

Does it has a way or necessary to convert a std::vector<bool> to bool*?

  virtual void Put(const T* src, int num_values) = 0;
  virtual void Put(const std::vector<T>& src, int num_values = -1);
  virtual void PutSpaced(const T* src, int num_values, const uint8_t* valid_bits,
                         int64_t valid_bits_offset) = 0;

Reply:
Ah, fair enough. Let's keep it like that, then, or use `unique_ptr`.